llvm/lib/SYCLLowerIR/LowerInvokeSimd.cpp

```diff
@@ -364,8 +364,10 @@ bool processInvokeSimdCall(CallInst *InvokeSimd,
     Function *InvokeSimdF = InvokeSimd->getCalledFunction();
     assert(InvokeSimdF && "Unexpected IR for invoke_simd");
     // - type of the obsolete (unmodified) helper:
-    Type *HelperArgTy = InvokeSimdF->getArg(HelperArgNo)->getType();
-    unsigned AS = dyn_cast<PointerType>(HelperArgTy)->getAddressSpace();
+    PointerType *HelperArgTy =
+        dyn_cast<PointerType>(InvokeSimdF->getArg(HelperArgNo)->getType());
+    assert(HelperArgTy && "Unexpected IR for invoke_simd");
```

Comment:
```suggestion
    PointerType *HelperArgTy =
        cast<PointerType>(InvokeSimdF->getArg(HelperArgNo)->getType());
```

`cast` already contains the required assertion and it doesn't have overhead on `if (isa<PointerType>)` which is present in `dyn_cast`

Reply:
+1

Reply:
Right, I'll do this additional fix. Thank you.